add action to translate readme

pelikhan opened the pull request:
This pull request configure AI-generated translations for french (only lessons and readme) using github actions and github models. Large files are still a bit of a problem, good start.
- the github action triggers on main when markdown files are modified.
- it performs as much translations as possible and pushes the result into main

pelikhan committed:
add action to translate readme

pelikhan committed:
feat: add French README for generative AI course

pelikhan committed:
fix: update translation workflow to correctly handle README.md and JSON files

pelikhan committed:
fix: correct file pattern for translation workflow to include nested JSON files

pelikhan committed:
Merge remote-tracking branch 'origin/main' into pelikhan/translations

pelikhan committed:
add french translations

pelikhan committed:
Add French translations for lessons

pelikhan committed:
fix: correct path in translation script and add usage data for new lessons

pelikhan committed:
Add French translation for Lesson 7: Model Context Protocol (MCP)

pelikhan committed:
fix: update translation workflow to include lesson README files and remove unnecessary branch

pelikhan reviewed (commented):
First iteration added. Some of the prompts are reaching 16k tokens and hitting the upper limit. Chunking documents will certainly help with this.

pelikhan committed:
Add French translations for additional lessons and update usage data

sinedied commented on lessons/01-intro-to-genai/translations/README.fr.md:
Is there a way to exclude specific words to be translated? Like for example "prompt"? Do you think we can use frontmatter metadata for that?

sinedied reviewed (commented):
Reviewed french translations and it's quite good!

Few questions though:
- Is there a global config file that we can use to set global translations rules, like for example avoid translating specific words?
- Is the cache shared between file, so an exact same paragraph in different files would be translated the same, for consistency? (and less token usage 🙂 )

sinedied commented on lessons/02-first-ai-app/translations/README.fr.md:
```diff
@@ -0,0 +1,501 @@
+
+Consultez l'[application d'exemple](/app/README.md) pour commencer.
+
+> \[!ASTUCE]
```
This [!TIP] should not be translated. A good candidate for global rules

pelikhan replied:
all github alerts should be fixed now.

sinedied commented on lessons/04-structured-output/translations/README.fr.md:
```diff
@@ -0,0 +1,561 @@
+
+## Récit - De Charybde en Scylla
+
+> > [!NOTE]
```
Double `>` here, not sure why? maybe trying to recompile the MD node after translation to see the same node structure was kept might help with issues like these?

pelikhan committed:
update some translations

sinedied commented on lessons/03-prompt-engineering/translations/README.fr.md:
Can these refusals be tagged in the translation cache?

pelikhan replied:
image alt text were not translated.

pelikhan committed:
better handling of alerts

pelikhan committed:
fix: correct formatting in French translations and update usage data for structured output lessons

pelikhan committed:
fix: correct formatting for notes and important sections in French translations

pelikhan committed:
translate image alt

pelikhan committed:
fix: add section for non-translatable technical terms in glossary

pelikhan committed:
Update French translations for RAG lesson and related files

- Revised the French translation of Ada's example in the RAG lesson to improve clarity and accuracy.
- Updated the corresponding JSON translation file to reflect the changes made in the RAG lesson.
- Added usage statistics for the RAG lesson translation in the usage log.

pelikhan committed:
fix: update CSV link and improve French translation for RAG lesson example

pelikhan commented:
> * Is there a global config file that we can use to set global translations rules, like for example avoid translating specific words?

The tool supports an "instructions.md" file that gets injected into the prompt.

> * Is the cache shared between file, so an exact same paragraph in different files would be translated the same, for consistency? (and less token usage 🙂 )

The cache is repo wide so if two strings match, they will have the same translation. We could also percolate a translation corpus in the future and use it as a template.

pelikhan committed:
fix: correct formatting for important notes in French translations
